doctor/types.py

```diff
@@ -468,99 +469,159 @@ def get_example(cls) -> list:
 
 
 class JsonSchema(SuperType):
-    """Represents a type loaded from a json schema."""
+    """Represents a type loaded from a json schema.
 
-    #: The description of what the type represents.  This is a placeholder
-    #: value, the actual value will come from your schema.
-    description = 'json type'
+    NOTE: This class should not be used directly.  Instead use
+    :func:`~doctor.types.json_schema_type` to create a new class based on
+    this one.
+    """
+    json_type = None
+    native_type = None
+    #: The loaded ResourceSchema
+    schema = None  # type: doctor.resource.ResourceSchema
     #: The full path to the schema file.
     schema_file = None  # type: str
     #: The key from the definitions in the schema file that the type should
     #: come from.
     definition_key = None  # type: str
 
```

Comment:
I had to move all the json schema loading/parsing code out since we need it loaded when we create the type using `json_schema_type`.  That function will load the schema and set it as an attribute on the generated class along with documentation attributes like 'example', 'description' and 'native_type'.  Also switched this to `__new__` so we can parse the value based on the expected json type.  That way query string parameters get cast to the appropriate type.